### PR TITLE
Add gzip support (with tests)

### PR DESCRIPTION
**Pull request by pwfisher**
@alanclarke Supersedes #31
Bumping version to 0.5.0 (from 0.3.2 in package.json) because 0.4.0 has been published.

### CODE REVIEW

**Comment by pwfisher** [comment]
Travis build fails for node 0.10 since zlib was added to Node core in 0.11. We should end support for Node 0.10 so we don't have to require a deprecated package, [kkaefer/DEPRECATED-node-zlib](https://github.com/kkaefer/DEPRECATED-node-zlib). I don't have permissions in Travis for this project, so my edits to `.travis.yml` would have no effect, please assist.

**Comment by arjansingh** [comment]
@alanclarke @doublerebel Any chance we could get this reviewed and released? I could really use this support in my project.

File: .travis.yml [hunk 1]
```diff
@@ -4,4 +4,3 @@ node_js:
   - "4"
   - "0.12"
   - "0.11"
-  - "0.10"
```
Comment (alanclarke) [inline .travis.yml]: Nice!

File: test/test-najax.js [hunk 2]
```diff
@@ -53,6 +54,16 @@ describe('url', function (next) {
...
 
+  function mockGzip (method) {
+    nock('http://www.example.com')[method]('/')
+      .reply(200, zlib.gzipSync('ok'), { 'Content-Encoding': 'gzip' })
+  }
+
+  function mockDeflate (method) {
+    nock('http://www.example.com')[method]('/')
+      .reply(200, zlib.deflateSync('ok'), { 'Content-Encoding': 'deflate' })
+  }
```
Comment (alanclarke) [inline test/test-najax.js]: Perfect

**Comment by alanclarke** [comment]
:+1:

**Pull request merged**